qt/applications/workbench/workbench/app/mainwindow.py

```diff
@@ -576,11 +577,18 @@ def closeEvent(self, event):
 
             # Kill the project recovery thread and don't restart should a save be in progress and clear out current
             # recovery checkpoint as it is closing properly
-            self.project_recovery.stop_recovery_thread()
-            self.project_recovery.closing_workbench = True
-            self.project_recovery.remove_current_pid_folder()
-
-            self.interface_manager.closeHelpWindow()
+            if self.project_recovery is not None:
+                self.project_recovery.stop_recovery_thread()
+                self.project_recovery.closing_workbench = True
+                self.project_recovery.remove_current_pid_folder()
+
+            # Cancel memory widget thread
+            if self.memorywidget is not None:
+                self.memorywidget.presenter.closing_workbench = True
```

Comment:
Can we avoid explicitly mentioning workbench? The widget lives in `mantidqt` and should not assume it's part of workbench.
Can the `cancel_memory_update` method not just some internal flag to mark it's closing rather than the outside needing to remember to do that?

Reply:
I have made it more generic and added it to `cancel_memory_update`.